google/cloud/aiplatform/metadata/schema/base_artifact.py

```diff
@@ -0,0 +1,134 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from google.auth import credentials as auth_credentials
+from google.cloud.aiplatform.metadata import artifact
+from google.cloud.aiplatform.metadata import constants
+from google.cloud.aiplatform.compat.types import artifact as gca_artifact
+from typing import Optional, Dict
+
+
+class BaseArtifactSchema(object):
+    """Base class for Metadata Artifact types.
+
+    This is the base class for defining various artifact types, which can be
+    passed to google.Artifact to create a corresponding resource.
+    Artifacts carry a `metadata` field, which is a dictionary for storing
+    metadata related to this artifact. Subclasses from ArtifactType can enforce
+    various structure and field requirements for the metadata field.
+
+    Args:
+        schema_title (str):
+            Optional. The schema title used by the Artifact, defaults to "system.Artifact"
+        resource_name (str):
+            Optional. The resource name of the Artifact following the format as follows.
+            This is globally unique in a metadataStore:
+            projects/123/locations/us-central1/metadataStores/<metadata_store_id>/artifacts/<resource_id>.
+        uri (str):
+            Optional. The uniform resource identifier of the artifact file. May be empty if there is no actual
+            artifact file.
+        display_name (str):
+            Optional. The user-defined name of the Artifact.
+        schema_version (str):
+            Optional. schema_version specifies the version used by the Artifact.
+            If not set, defaults to use the latest version.
+        description (str):
+            Optional. Describes the purpose of the Artifact to be created.
+        metadata (Dict):
+            Optional. Contains the metadata information that will be stored in the Artifact.
+        state (google.cloud.gapic.types.Artifact.State):
+            Optional. The state of this Artifact. This is a
+            property of the Artifact, and does not imply or
+            capture any ongoing process. This property is
+            managed by clients (such as Vertex AI
+            Pipelines), and the system does not prescribe or
+            check the validity of state transitions.
+        **kwargs:
+            Optional. Additional Args that will be passed directly to the Artifact base method for backward compatibility.
+    """
+
+    ARTIFACT_PROPERTY_KEY_RESOURCE_NAME = "resourceName"
+    SCHEMA_TITLE = "system.Artifact"
+
+    def __init__(
+        self,
+        schema_title: Optional[str] = None,
+        resource_name: Optional[str] = None,
+        uri: Optional[str] = None,
+        display_name: Optional[str] = None,
+        schema_version: Optional[str] = None,
+        description: Optional[str] = None,
+        metadata: Optional[Dict] = None,
+        state: gca_artifact.Artifact.State = gca_artifact.Artifact.State.LIVE,
```

Comment:
@jialuzh 

You recommended not defaulting to LIVE in the experiments PR. I think we should be consistent.

My preference, after using against the console, is to default to LIVE across because the artifacts appears grayed out in the console if not.

Reply:
@sasha-gitg Right, I also noticed the UI gray out issue yesterday during our mini bug batch yesterday and I also changed my mind and agree that we should revert it back to LIVE as it is used by UI. I was planning to sync with you on this as well and looks like we landed in the same conclusion already:) And yes, I agree we should do it across the board to be consistent.

Reply:
resolving this as seems like everyone is in agreement on defaulting to live.